### PR TITLE
detools: add version 0.53.0, suppport conan v2

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **detools/***

- support conan v2
- add package_type
- add version 0.53.0
- update dependencies
- replace zip to tar ball

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by valgur** [comment]
I will probably close #18659 in favor of this one. Thanks!

**Comment by toge** [comment]
@valgur 
Sorry!
I missed your PR.
Thanks a lot for passing me the right to update.

File: recipes/detools/all/CMakeLists.txt [hunk 1]
```diff
@@ -1,9 +1,20 @@
 cmake_minimum_required(VERSION 3.1)
```
Comment (jwillikers) [inline recipes/detools/all/CMakeLists.txt:1]: Can you bump this minimum required version to `3.15`?

File: recipes/detools/all/CMakeLists.txt [hunk 2]
```diff
@@ -1,9 +1,20 @@
...
-install(FILES source_subfolder/c/detools.h DESTINATION include)
+add_library(detools ${DETOOLS_SRC_DIR}/c/detools.c)
+target_link_libraries(detools PRIVATE
+    heatshrink::heatshrink
+    LibLZMA::LibLZMA
+)
+
+install(
+    TARGETS detools
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}
```
Comment (jwillikers) [inline recipes/detools/all/CMakeLists.txt:15]: Need to include `include(GNUInstallDirs)` to make the `CMAKE_INSTALL_` variables available.

File: recipes/detools/all/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -1,10 +1,7 @@
 cmake_minimum_required(VERSION 3.1)
```
Comment (jwillikers) [inline recipes/detools/all/test_package/CMakeLists.txt:1]: This min version can also be bumped.

**Review by jwillikers — CHANGES_REQUESTED** [review]

File: recipes/detools/all/CMakeLists.txt (hunk 1 above)
Comment (toge) [reply recipes/detools/all/CMakeLists.txt]: @jwillikers 
Thank you for your review!
I fixed these points.

**Review by jwillikers — APPROVED** [review]

File: recipes/detools/all/CMakeLists.txt (hunk 1 above)
Comment (jwillikers) [reply recipes/detools/all/CMakeLists.txt]: Thanks! LGTM

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**